### ISSUE TITLE
List previously submitted AWS Batch jobs in the job queue

### ISSUE THREAD

**Issue by huddlej** [nextstrain/cli#125]
## Context

When a user submits a job to AWS Batch, they receive a job id they can use to reattach to the job later on. If the user doesn't copy this id somewhere (a lab notebook, etc.) and they do not know how to use the AWS web interface, then they've effectively lost the AWS Batch job.

## Description

Users need a way to list their previously submitted jobs.

## Possible solution

The CLI could provide an interface for users to list jobs they have previously submitted, providing the job id required to reattach and any other metadata that is generated by the initial submission (at least submission date) so users can figure out which job is which.

This interface seems to be outside of the scope of `nextstrain build` and should probably be its own subcommand.

Alternately, the CLI could send an email with job status to the user much like some HPC systems do for new cluster submissions.

**Comment by trvrb** [comment]
I've struggled with a related issue in that AWS Batch has a relatively short window in which it remembers your job ID. This makes it so that if I come back to a completed run ~2 days later I can no longer attach to a completed run and download build artifacts.

However, the compressed artifacts file in the `nextstrain-jobs` S3 bucket persists for ~30 days. I often end up deciphering S3 file name by figuring out when a job completed to be able to download the correct file from S3 (because I missed the ~2 day window that the job results were available through their Batch ID).

For Katie's use case, she should have access to https://console.aws.amazon.com/batch/ to list running, failed and completed jobs, etc...

**Comment by tsibley** [comment]
@huddlej Yeah, this would be good. Biggest issue here is a good UI/UX, I think, though implementation is complicated by Trevor's point in that this feature probably doesn't want to rely on _just_ the state kept by Batch itself.

The related issue Trevor brings up of not being able to re-attach to an old completed job can be addressed by keeping some minimal local state and is described in more detail in https://github.com/nextstrain/cli/issues/113.

I think both this issue and #113 should use the same local job state. Once we have local state, we have to answer questions like, "When does a job get removed from local state?" or alternatively "When does a job present in local state stop being listed (by default)?"

One answer I think is promising is to keep a minimal log of submitted job information that's never pruned, and for listing of jobs, apply a default filter on time/recency (maybe with the option to show the whole list or adjust the time range, or maybe just document where to go spelunking yourself).
